test/Spec/MockContract.hs

```diff
@@ -295,7 +310,27 @@ mockCallCommand ShellArgs {cmdName, cmdArgs, cmdOutParser} = do
       pure $ cmdOutParser ""
     ("cardano-cli", "transaction" : "submit" : _) ->
       pure $ cmdOutParser ""
-    _ -> throwError @Text "Unknown command"
+    (unsupportedCmd, unsupportedArgs) ->
+      throwError @Text
+        ("Unsupported command: " <> Text.intercalate " " (unsupportedCmd : unsupportedArgs))
+
+mockQueryTip :: forall (w :: Type). MockContract w String
+mockQueryTip = do
+  state <- get @(MockContractState w)
+
+  let slot = Text.pack $ show $ getSlot $ tipSlot (state ^. tip)
+      blockId = decodeUtf8 $ getBlockId $ tipBlockId (state ^. tip)
+      blockNo = Text.pack $ show $ unBlockNumber $ tipBlockNo (state ^. tip)
```

Comment:
Maybe unpack the tip before hand to avoid repition
Can use namepuns like
```suggestion
  let Tip {tipSlot, tipBlockId, tipBlockNo} = state ^. tip
      slot = Text.pack $ show $ getSlot tipSlot
      blockId = decodeUtf8 $ getBlockId tipBlockId
      blockNo = Text.pack $ show $ unBlockNumber tipBlockNo
```

Reply:
The problem is the same here, Tip is a sum type, so I realised that the record functions are partial here. Used a good old case statement instead.